src/m365/aad/commands/approleassignment/approleassignment-remove.ts

```diff
@@ -0,0 +1,248 @@
+              // the role specified in the scope option does not belong to the found service principles
+              // throw an error and show list with available roles (scopes)
+              let availableRoles: string = '';
+              appRolesFound.map((r: AppRole) => availableRoles += `${os.EOL}${r.value}`);
+
+              return Promise.reject(`The scope value '${scope}' you have specified does not exist for ${args.options.resource}. ${os.EOL}Available scopes (application permissions) are: ${availableRoles}`);
+            }
+
+            appRolesToBeDeleted.push(existingRoles[0]);
+          }
+          const tasks: Promise<any>[] = [];
+
+          for (const appRole of appRolesToBeDeleted) {
+            const appRoleAssignment = sp.appRoleAssignments.filter((role: AppRoleAssignment) => role.appRoleId === appRole.id);
+            if (!appRoleAssignment.length) {
+              return Promise.reject('App role assignment not found');
+            }
+            tasks.push(this.removeAppRoleAssignmentForServicePrincipal(sp.id, appRoleAssignment[0].id));
+          };
+
+          return Promise.all(tasks);
+        }).then((): void => {
+          if (this.verbose) {
+            logger.log(chalk.green('DONE'));
+          }
+
+          cb();
+        }, (res: any): void => this.handleRejectedODataJsonPromise(res, logger, cb));;
+    }
+
+    if (args.options.confirm) {
+      removeAppRoleAssignment();
+    }
+    else {
+      Cli.prompt(
+        {
+          type: 'confirm',
+          name: 'continue',
+          default: false,
+          message: `Are you sure you want to remove the appRoleAssignment with scope ${args.options.scope} for resource ${args.options.resource}?`
+        },
+        (result: { continue: boolean }): void => {
+          if (!result.continue) {
+            cb();
+          }
+          else {
+            removeAppRoleAssignment();
+          }
+        }
+      );
+    }
+  }
+
+  private getServicePrincipalForApp(filterParam: string): Promise<{ value: ServicePrincipal[] }> {
+    const spRequestOptions: any = {
+      url: `${this.resource}/v1.0/servicePrincipals?$expand=appRoleAssignments&$filter=${filterParam}`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request.get<{ value: ServicePrincipal[] }>(spRequestOptions);
+  }
+
+  private removeAppRoleAssignmentForServicePrincipal(spId: string, appRoleAssignmentId: string): Promise<ServicePrincipal> {
+    const spRequestOptions: any = {
+      url: `${this.resource}/v1.0/servicePrincipals/${spId}/appRoleAssignments/${appRoleAssignmentId}`,
+      headers: {
+        'accept': 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request.delete(spRequestOptions);
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --appId [appId]',
```

Comment:
Doesn't match the spec. We haven't defined `-i` because it's ambiguous between `appId` and `objectId`